### PR TITLE
client: prepare InstallSystemOptions for real use

### PR DESCRIPTION
**Pull request by mvo5**
This commit prepares the `InstallSystemOptions` to be ready for real use inside the installer API. This requires two things:

**1. Create client.InstallVolume{,Struct}**

The new struct is similar to `gadget.Volume{,Struct}` but tailored towards the installer use-case. The `client.VolumeStructure` contains an extra `Device` and `UnencryptedDevice` field that is filled in from the installer.

**2. Fix type mismatch in `client.InstallSystemOptions.OnVolumes`**

The volume information in snapd is usually encoded as: `map[string]*gadget.Volume`. However due to an oversight in the client code (and because it was not used yet) a small mistake made it into the code and it got encoded as:
`map[string][]gadget.Volume` there.

### CODE REVIEW

File: client/systems.go [hunk 1]
```diff
@@ -171,14 +171,28 @@ const (
 	InstallStepFinish                 InstallStep = "finish"
 )
 
+type InstallVolumeStructure struct {
+	*gadget.VolumeStructure
+
+	// The installer need to set those
```
Comment (pedronis) [inline client/systems.go:177]: ... those as needed depending on step.

File: client/systems_test.go [hunk 2]
```diff
@@ -324,21 +324,46 @@ func (cs *clientSuite) TestRequestSystemInstallEmptySystemLabel(c *check.C) {
 	c.Check(cs.req, check.IsNil)
 }
 
-func (cs *clientSuite) TestRequestSystemInstallHappy(c *check.C) {
+func (cs *clientSuite) TestRequestSystemInstallShadowsExistingStruct(c *check.C) {
```
Comment (pedronis) [inline client/systems_test.go:327]: I think the test should have kept the previous name, there is anyway a comment inside about the shadowing issue

File: client/systems_test.go [hunk 3]
```diff
@@ -324,21 +324,46 @@ func (cs *clientSuite) TestRequestSystemInstallEmptySystemLabel(c *check.C) {
...
-			{
+			Volume: &gadget.Volume{
 				Schema:     "dos",
 				Bootloader: "mbr",
-				ID:         "0c",
+				ID:         "id",
 				// Note that name is not exported as json
 				Name: "pc",
+				// Note that this will be "shadowed"
+				// and not actually be visible, see
```
Comment (pedronis) [inline client/systems_test.go:343]: ... visible, as it is more nested, see ...

**Review by pedronis — APPROVED** [review]
thanks, some small remarks

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM

**Pull request merged**